builtin/providers/aws/resource_aws_route53_delegation_set.go

```diff
@@ -0,0 +1,111 @@
+package aws
+
+import (
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/service/route53"
+)
+
+func resourceAwsRoute53DelegationSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRoute53DelegationSetCreate,
+		Read:   resourceAwsRoute53DelegationSetRead,
+		Delete: resourceAwsRoute53DelegationSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"reference_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"name_servers": &schema.Schema{
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRoute53DelegationSetCreate(d *schema.ResourceData, meta interface{}) error {
+	r53 := meta.(*AWSClient).r53conn
+
+	callerRef := resource.UniqueId()
+	if v, ok := d.GetOk("reference_name"); ok {
+		callerRef = strings.Join([]string{
+			v.(string), "-", callerRef,
+		}, "")
+	}
+	input := &route53.CreateReusableDelegationSetInput{
+		CallerReference: aws.String(callerRef),
+	}
+	if v, ok := d.GetOk("zone_id"); ok {
```

Comment:
So this was actually in the original concept, but then I trashed the idea of defining `zone_id` directly from this resource as I can't see any meaningful use-case for it.

There's `aws_route53_zone.delegation_set_id` which I'm adding and I believe the pairing will be much more useful that way.

So I guess I can just remove these three lines unless you can think of a use-case.